lib/bolt/transport/remote.rb

```diff
@@ -34,7 +34,7 @@ def run_task(target, task, arguments, options = {})
         remote_task = task.remote_instance
 
         result = transport.run_task(proxy_target, remote_task, arguments, options)
```

Comment:
Wont the transports already give a Result object? Why do we construct a new one?

Reply:
The result is the result from the proxy, rather than the specified target.

It looks like we're creating a new `Result` so that it has the right target specified.